base/loading.jl

```diff
@@ -145,7 +145,7 @@ function show(io::IO, ex::PrecompilableError)
     end
 end
 precompilableerror(ex::PrecompilableError, c) = ex.isprecompilable == c
-precompilableerror(ex::LoadError, c) = precompilableerror(ex.error, c)
+precompilableerror(ex::Union{LoadError,InitError}, c) = precompilableerror(ex.error, c)
 precompilableerror(ex, c) = false
```

Comment:
Not sure if this is useful for other cases, but this change kind of begs for `abstract WrappedException <: Exception`